org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -30,6 +30,19 @@
  * QueryExecutor implementation for the V3 protocol.
  */
 public class QueryExecutorImpl implements QueryExecutor {
+
+    /**
+     * Prepare UTF8-encoded values for $1, $2, etc, so we do not spend time
+     * and memory when sending prepare command.
```

Comment:
> Prepare UTF8-encoded values for $1, $2, etc, so we do not spend time and memory when sending prepare command.

Doesn't quite read right ... How about this instead:

    Cache UTF8-encoded values for $1, $2, etc, so we do not have to repeatedly generate them for prepared statements.